docs/source/examples/vqe-demo_custom-gate.myst

```diff
@@ -0,0 +1,282 @@
+
+    freq = Counter(bitstring_tuples)
+
+    
+    exp_val = 0
+    for bitstring, count in freq.items():
+        bitstring_int = int("".join([str(x) for x in bitstring[::-1]]), 2)
+        if parity_even_p(bitstring_int, marked_qubits):
+            exp_val += float(count) / samples
+        else:
+            exp_val -= float(count) / samples
+    return exp_val
+```
+
+Calculate the parity of elements at indexes in marked_qubits
+The function is a modified version of `parity_even_p` from 
+https://github.com/rigetti/grove/blob/master/grove/pyvqe/vqe.py.
+
+```{code-cell} ipython3
+def parity_even_p(state, marked_qubits):
+    mask = 0
+    for q in marked_qubits:
+        mask |= 1 << q
+    return bin(mask & state).count("1") % 2 == 0
+```
+
+Scan over the parameter theta and plot energy expecation, without mitigation
+
+```{code-cell} ipython3
+thetas = np.linspace(0, 2 * np.pi, 51)
+results = []
+
+for theta in thetas:
+    results.append(executor(theta, backend, readout_qubit, samples, hamiltonian, program))
+```
+
+Run VQE routine without noise mitigation
+
+```{code-cell} ipython3
+init_angle = [3.0]
+
+res = optimize.minimize(
+    executor,
+    init_angle,
+    args=(backend, readout_qubit, samples, hamiltonian, program),
+    method="Nelder-Mead",
+    options={"xatol": 1.0e-3, "fatol": 1.0e-2},
+)
+print(res)
+```
+
+Now introduce noise mitigation technique Zero Noise Extrapolation
+
+Create new objective function to calculate expecation value w/ noise mitigation
+
+```{code-cell} ipython3
+def mitigated_expectation(
+    thetas, backend, readout_qubit, samples, pauli_sum, executable: Program, factory
+) -> float:
+    mitigated_exp = zne.execute_with_zne(
+        executable,
+        lambda p: executor(thetas, backend, readout_qubit, samples, pauli_sum, p),
+        factory,
+        fold_gates_from_right,
+    )
+    return mitigated_exp
+```
+
+Here we use a linear technique for the Zero Noise Extrapolation
+
+```{code-cell} ipython3
+fac = mitiq.zne.inference.LinearFactory(scale_factors=[1.0, 3.0])
+```
+
+Scan over the parameter theta and plot energy expecation with noise mitigation
+
+```{code-cell} ipython3
+results_zne = []
+for theta in thetas:
+    results_zne.append(
+        mitigated_expectation(theta, backend, readout_qubit, samples, hamiltonian, program, fac)
+    )
+
+_ = plt.figure()
+_ = plt.plot(thetas, results, "o-", label="Noisy landscape")
+_ = plt.plot(thetas, results_zne, "o-", label="Mitigated landscape")
+_ = plt.xlabel(r"$\theta$", fontsize=18)
+_ = plt.ylabel(r"$\langle \Psi(\theta) | Z | \Psi(\theta) \rangle$", fontsize=18)
+_ = plt.title("Mitigated Energy Landscape")
+plt.show()
+```
+
+Run VQE routine with noise mitigation
+
+```{code-cell} ipython3
+res_zne = optimize.minimize(
+    mitigated_expectation,
+    init_angle,
+    args=(backend, readout_qubit, samples, hamiltonian, program, fac),
+    method="Nelder-Mead",
+    options={"xatol": 1.0e-3, "fatol": 1.0e-2},
+)
+print(res_zne)
```

Comment:
After this block, please conclude the example with an analysis of results and a short conclusion.

Reply:
I added some explanations around the results and a short conclusion section at the end.